addons/mail/__manifest__.py

```diff
@@ -94,6 +94,7 @@
         'data/ir_config_parameter_data.xml',
         'data/res_partner_data.xml',
         'data/mail_message_subtype_data.xml',
+        'data/mail_templates_channel_invitation.xml',
```

Comment:
actually ``mail_templates_chatter.xml`` file holds templates like your new one (templates used for misc notifications / emails / messages).